Version 13 fixes v13-19-0

e-lobo opened the pull request:
Please Review and merge:
1. https://github.com/leam-tech/renovation_core/pull/81 ✅

Changes include but not limited to:
1. [Fix](https://github.com/leam-tech/frappe/pull/9/commits/f9c1872e7538397b578f7bdd015c68e1a84ba0b2) for is_owner 
2. [Fix](https://github.com/leam-tech/frappe/pull/9/commits/7ef73c324010fd1447d5b770a21f1fc824c1cb73) for test_runner when we run with flag --doctype
3. [Fix](https://github.com/leam-tech/frappe/pull/9/commits/a86e53f7e9360b7ae1698de7235b7f509b335d3a) validate_link for child tables
4. [Fix](https://github.com/leam-tech/frappe/pull/9/commits/830fc9c98a3acaaf665b380d7798f15196d6d865) import file ..
this is [fixed](https://github.com/frappe/frappe/commit/562a192e43a6a4d6fceb5a455ed76c796a96a55c)

⚠️ Warning: Please checkout pr mentioned above before updating if `renovation_core` is installed ⚠️

navdeepghai committed:
feat(fixes): for frappeclient post_api(POST) request being redirected to GET request on calling server

(cherry picked from commit d73428df919dc98e7ee13af92af5fd8a66da66d4)

pateljannat committed:
fix: discussions template and time

(cherry picked from commit 8aa977c5f02bfc32c3e8c609f51aab4e3c4cb00b)

# Conflicts:
#	frappe/public/js/frappe-web.bundle.js
#	frappe/public/scss/website/index.scss
#	frappe/templates/base.html

pateljannat committed:
fix: sider

(cherry picked from commit cd967b36b8f6a11fcee186ce1ece6be62b7e83e5)

pateljannat committed:
fix: sider

(cherry picked from commit 7649988efcd8c939f54ed380badfb50ca2ce9535)

pateljannat committed:
test: ui

(cherry picked from commit dc73dea31838cc599a0b820c6a88dd6c06caebd9)

# Conflicts:
#	frappe/tests/ui_test_helpers.py

pateljannat committed:
fix: sider

(cherry picked from commit 6f64822a55dfb852181ed6abb6691bf691dfe2c6)

pateljannat committed:
fix: increased wait time

(cherry picked from commit 4138d876e2175f8e399f4405d260157b65a47c5f)

pateljannat committed:
fix: removed unnecessary redirect

(cherry picked from commit afdf5611401dcc30b0750d4582fd262519915692)

pateljannat committed:
fix: web page and load time

(cherry picked from commit d49ce24e097cdc68cf8a7a7fb77ffe9269c6b040)

pateljannat committed:
fix: tests

(cherry picked from commit bc5fcce14f060616b5f05912f1e0e326fda10594)

pateljannat committed:
fix: reply seperator

(cherry picked from commit 367099f1faf0e9dd482b024dcb0c6bc59473f368)

pateljannat committed:
fix: label for web page in template

(cherry picked from commit bd184ccec35addf95d61e2c5ef6aacc17b58af02)

pateljannat committed:
fix: scroll, translations, naming series, placeholder and variable definition

(cherry picked from commit 47fab75bcd68581e82dfba88e00a3d6f4a2151f4)

gavindsouza committed:
chore: Bump psycopg2-binary Python client

(cherry picked from commit 39d63641f3d9928b2648ec83e793badf5c92f544)

pateljannat committed:
style: indentation, etc

(cherry picked from commit fd57185bc9376d1dc7ee0c2656829e985c65a3e3)

# Conflicts:
#	frappe/public/scss/website/index.scss
#	frappe/tests/ui_test_helpers.py

rohitwaghchaure committed:
feat: add new row in the grid using keyboard keys

(cherry picked from commit 6111cc8db9d1079544c457e53835f668535353b3)

rohitwaghchaure committed:
test: added test cases to validate keyboard shortcuts for the grid

(cherry picked from commit 4e3f996d642e1b14ca2732a05552490327be1f27)

rohitwaghchaure committed:
feat: added shortcut keys description for grid

(cherry picked from commit 073020c69d995cbdfe9daa7f738e294b34f13a81)

shariquerik committed:
fix: show unique records in list view

(cherry picked from commit aea21c2fe68bcd5ba8700f2cf62ff12881a63526)

surajshetty3416 committed:
ci: Use node version 14 to avoid node-sass failure in patch testing build (#15176)

(cherry picked from commit 8a84ae4f0812d2dc5700b7e7011fa96141be204f)

assemmarwan commented on frappe/test_runner.py:
```diff
@@ -73,7 +74,9 @@ def main(app=None, module=None, doctype=None, verbose=False, tests=(),
-		frappe.utils.scheduler.enable_scheduler()
+		if cint(frappe.db.get_value("System Settings", fieldname="enable_scheduler")):
+			frappe.utils.scheduler.disable_scheduler()
+			if frappe.db: frappe.db.commit()
```
@e-lobo Isn't this redundant?

e-lobo replied:
@assemmarwan this is to ensure any write operations in db are completed before tests start to run.. there is an issue if we remove frappe.db.commit() when we run tests with --doctype

e-lobo replied:
@assemmarwan fixed.. it was in the wrong place 👍

pateljannat committed:
fix: conflict in scss

pateljannat committed:
fix: conflicts in template scripts

pateljannat committed:
fix: conflicts in ui test helper

pateljannat committed:
fix: removed include_script from base.html

surajshetty3416 committed:
Merge pull request #15206 from frappe/mergify/bp/version-13-hotfix/pr-14226

mergify[bot] committed:
Merge pull request #15177 from codescientist703/remove-chat

chore: Remove chat module (Backport)

mergify[bot] committed:
fix: ReDoS bug: GHSL-2021-122 (backport #15201) (#15216)

Co-authored-by: Kevin Backhouse <kevinbackhouse@github.com>

ankush committed:
fix: missing content in communication composer

`message` is used as args in some places and `txt` in another.
Supporting both since there's no clear interface to this arguments for
this component.

(cherry picked from commit f589342c7061f6eefca30534676712fd0995b1f5)

ankush committed:
feat: copy error to clipboard

Currently error can be reported if email is setup but other than that
only way to find error info is use browser's console.

This change adds a secondary button to copy error info to clipboard.

(cherry picked from commit 5b6012d0d3944153bb3bb8156c21482a1a8583da)

ankush committed:
fix: copy to clipboard strips off newlines

input element isn't suitable for this hack, switch to textarea for
supporting newlines in text to be copied.

(cherry picked from commit ecdb8932864ced64c9969fe351b492cabdfd10da)

ankush committed:
refactor: use navigator api for clipboard

(cherry picked from commit f3cbaf9df782b16558d9ea7a6811c8ae71c84602)

ankush committed:
fix: disable data import legacy on v13 (#15232)

kennethsequeira committed:
fix: valid parent doctype error (#15233)

Co-authored by: Ankush Menat <me@ankush.dev>

(cherry picked from commit bac62ea014a01d013f8abf502a3ce0481351f84f)

mergify[bot] committed:
Merge pull request #15234 from frappe/mergify/bp/version-13-hotfix/pr-15233

fix: valid parent doctype error (backport #15233)

rohitwaghchaure committed:
fix: record name creating based on naming seires and not as per the use set

(cherry picked from commit 7118fb2e815e6f8567b81e7a0017b2129c9358d1)

rohitwaghchaure committed:
Merge pull request #15241 from frappe/mergify/bp/version-13-hotfix/pr-15238

fix: record name creating based on naming series and not as per the user input (backport #15238)

nemesis189 committed:
fix: Sales invoice cancellation error due to get_list (#15243)

* fix: Sales invoice cancellation error due to get_list

* fix: missed out one `get_list`

(cherry picked from commit 2b8328715e1c084a94c9ddcdccc6a12c8551eebf)

mergify[bot] committed:
fix: update modified of Singles (#15251)

(cherry picked from commit 2ab030c4d813decdc19ecf43bba6601e34a7d721)

Co-authored-by: barredterra <14891507+barredterra@users.noreply.github.com>

mergify[bot] committed:
fix: update modified of Singles (#15252)

(cherry picked from commit 2ab030c4d813decdc19ecf43bba6601e34a7d721)

Co-authored-by: barredterra <14891507+barredterra@users.noreply.github.com>

mergify[bot] committed:
Merge pull request #15254 from frappe/mergify/bp/version-13-pre-release/pr-15176

ci: Use node version 14 to avoid node-sass failure in patch testing build (backport #15176)

phot0n committed:
fix: not able to go back from user profile page

credit: @ankush
(cherry picked from commit 41bb12a2065a7065fa5fdb63f887f3fa64f95b42)

surajshetty3416 committed:
Merge pull request #15245 from frappe/mergify/bp/version-13-hotfix/pr-15243

shridarpatil committed:
fix: Make query Postgres compatible (#15240)

* fix: updated query compatibale with postgres

* fix: updated query compatible with postgres

(cherry picked from commit 6acbfd5e7466141b134db6091e33d17d74e136e6)

surajshetty3416 committed:
Merge pull request #15266 from frappe/mergify/bp/version-13-hotfix/pr-15240

mergify[bot] committed:
fix: Make query Postgres compatible (backport #15239) (#15267)

Co-authored-by: Shridhar Patil <shridhar.p@zerodha.com>

ankush committed:
fix: remove bench path from traceback

This is mostly useless for debugging and clutters the error message.

(cherry picked from commit 38cd12a95b33fae974e4227e62babfbf3f70e998)

ankush committed:
fix: change route when browsing someone else's profile

(cherry picked from commit faa8c371fb89534777beb37f34317ae33d69c2de)

ankush committed:
fix: leaderboard link in sidebar

(cherry picked from commit a7a05701780f242ff3b562ccae9561e81c87f381)

ankush committed:
fix: remove slash prefix

(cherry picked from commit 00d38d568c0f5f82cca908a08421b1d4d32906b5)

surajshetty3416 committed:
Merge pull request #15269 from frappe/mergify/bp/version-13-hotfix/pr-15265

Co-authored-by: phot0n <ritwikpuri5678@gmail.com>
Co-authored-by: Ankush Menat <ankush@frappe.io>

surajshetty3416 committed:
Merge pull request #15270 from frappe/mergify/bp/version-13-hotfix/pr-15268

surajshetty3416 committed:
Merge pull request #15235 from frappe/mergify/bp/version-13-hotfix/pr-15225

vrslev committed:
fix(deps): Drop svg-sprite to fix linux/arm64 build

mergify[bot] committed:
Merge pull request #15275 from vrslev/fix-arm64-build-2

fix(deps): Remove svg-sprite to fix linux/arm64 build

hasnain2808 committed:
revert: set amended docname to original docname-1

Making use_original_name_for_amended_document in system setting to be readonly. This is done to make sure others cannot turn on the setting

hasnain2808 committed:
revert: set amended docname to original docname-1

Making use_original_name_for_amended_document in system setting to be readonly. This is done to make sure others cannot turn on the setting

(cherry picked from commit 1911b59d3a0ed51f6410b36832f3ea2ca81694ee)

Commit:
fix: remove display none class

(cherry picked from commit 66febd024ab0e698b5006dbb22a9124ec38edc76)

Commit:
chore: add icon classes for each event type

(cherry picked from commit 65dfd87f49a857cdc4a8f82d490c43820dbb33a6)

rohitwaghchaure committed:
chore: merge branch 'version-13-hotfix' into 'version-13-pre-release'

barredterra committed:
fix: import doc / fixtures

(cherry picked from commit c92e6ab85d5cbe06713e7ad23c49698c9be3c99c)

barredterra committed:
fix: import doc / fixtures

(cherry picked from commit c92e6ab85d5cbe06713e7ad23c49698c9be3c99c)

rohitwaghchaure committed:
Merge pull request #15286 from rohitwaghchaure/merge-hotfix-to-pre-release-for-13-17

chore: merge branch 'version-13-hotfix' into 'version-13-pre-release'

surajshetty3416 committed:
Merge pull request #15289 from frappe/mergify/bp/version-13-hotfix/pr-15284

surajshetty3416 committed:
Merge pull request #15290 from frappe/mergify/bp/version-13-hotfix/pr-15279

mergify[bot] committed:
Merge pull request #15278 from hasnain2808/make-amended-doc-orignal-name-setting-readonly-v13

fix: Make "Use original name for amended document" setting read-only

surajshetty3416 committed:
Merge pull request #15291 from frappe/mergify/bp/version-13-pre-release/pr-15278

akazyti committed:
fix: Error code for "user not found" (#15223)

* fix: error code for not found

* fix: updating test

* fix: update status code

* fix: update test for status code 400

(cherry picked from commit fbf926d1f66d6a5e1d7797ba8987626016bba913)

surajshetty3416 committed:
Merge pull request #15292 from frappe/mergify/bp/version-13-hotfix/pr-15223

shariquerik committed:
fix: refresh all data

(cherry picked from commit f26b0234d669e36daaa67b912a50e5b71257f62c)

surajshetty3416 committed:
fix: Combine request args and form data

- To accept argument via both way in a single request

(cherry picked from commit fba15b3874f48e3863de1a3ecf06edd9ef9bfa67)

surajshetty3416 committed:
Merge pull request #15294 from frappe/mergify/bp/version-13-hotfix/pr-15293

surajshetty3416 committed:
Merge pull request #15297 from frappe/mergify/bp/version-13-pre-release/pr-15288

surajshetty3416 committed:
Merge pull request #15296 from frappe/mergify/bp/version-13-hotfix/pr-15288

shariquerik committed:
refactor: reset start and page length after every refresh

(cherry picked from commit d611c01180ec0ed0ea95be10ead7879c5be8e9c4)

shariquerik committed:
chore: removed unused code

(cherry picked from commit 54adcf14aaea6c001405826f9eb8b64db9050405)

ruchamahabal committed:
fix: select perm not visible in role permission summary dialog

(cherry picked from commit 9a5804c4e89a346f87edeba18786eb85f5a97d4c)

vrslev committed:
fix(Translations): Timeline added/removed rows fieldname

ankush committed:
Merge pull request #15301 from vrslev/fix-timeline-translations-2

fix(Translations): Timeline added/removed rows fieldname

ankush committed:
Merge pull request #15303 from frappe/mergify/bp/version-13-hotfix/pr-15300

fix: select perm not visible in role permission summary dialog (backport #15300)

vrslev committed:
fix(Translations): Search suggestions for lists, reports etc

surajshetty3416 committed:
fix: Move barcode style to scss file

So that it applies the style gets applied only in form view and not in printview

(cherry picked from commit a74ce1dba7b61b763ce55d00e76402499155401b)

surajshetty3416 committed:
fix: Move barcode style to scss file

So that it applies the style gets applied only in form view and not in printview

surajshetty3416 committed:
Merge pull request #15307 from surajshetty3416/fix-barcode-rendering

nextchamp-saqib committed:
fix(ux): always show scan barcode button

(cherry picked from commit 62a1f6586d0be127206a2a8118b6939d9373d4cf)

# Conflicts:
#	frappe/public/js/frappe/form/controls/data.js

surajshetty3416 committed:
Merge pull request #15308 from frappe/mergify/bp/version-13-pre-release/pr-15307

shariquerik committed:
fix: Load 20 more rows on click of 'Load More'

(cherry picked from commit 6c938d99aef52b4d5fe0f7e9a42662d54c88ae55)

surajshetty3416 committed:
Merge pull request #15310 from frappe/mergify/bp/version-13-hotfix/pr-15175

surajshetty3416 committed:
fix: Resolve conflicts

surajshetty3416 committed:
Merge pull request #15313 from frappe/mergify/bp/version-13-hotfix/pr-15309

pateljannat committed:
fix: datetime in build

(cherry picked from commit ab811415c232117e96d8643b0a09ceb192dbee5a)

pateljannat committed:
fix: datetime in build

pateljannat committed:
fix: using format_datetime instead of prettydate

pateljannat committed:
fix: using format_datetime instead of prettydate

(cherry picked from commit 74801b05e98ee55b8982907aa478d9337897a511)

pateljannat committed:
fix: removed unnecessary data attributes and class

(cherry picked from commit 850799a33982dc23a88a32db2a466a5103e600f9)

pateljannat committed:
fix: removed unnecessary data attributes and class

surajshetty3416 committed:
Merge pull request #15314 from frappe/fix-disucssions-component

surajshetty3416 committed:
Merge pull request #15316 from frappe/mergify/bp/version-13-pre-release/pr-15314

surajshetty3416 committed:
fix: Make description translatable

(cherry picked from commit ede927f03e12dff4643f60ee478c366e9b346fda)

surajshetty3416 committed:
fix: Update descriptions

(cherry picked from commit 79c2464a7a28d141695f7298fc04144cf07fa593)

sagarvora committed:
feat: translations in Data Import

(cherry picked from commit 5df970013ebd2eaad2d9b88299938ecc3e8cb722)

ankush committed:
fix: set link field to `undefined` instead of empty string

This breaks other code where undefined values are removed, technically
invalid value for link field => unset link field so it should be
undefined

(cherry picked from commit b633f31bf4b3e70c71adad71fba5838fbba0d7de)

ankush committed:
fix: set link field to `undefined` instead of empty string

This breaks other code where undefined values are removed, technically
invalid value for link field => unset link field so it should be
undefined

(cherry picked from commit b633f31bf4b3e70c71adad71fba5838fbba0d7de)

sagarvora committed:
fix: remove unnecessary loop

(cherry picked from commit a952c8994d2f9665a05b96eab03e1e8f025bf331)

sagarvora committed:
fix: remove unnecessary translation

(cherry picked from commit e6d550b45199a576c613eae9f6a801f9ccdb3089)

surajshetty3416 committed:
refactor:  Use null instead of undefined

(cherry picked from commit 1e3d58c14f614ecac1eb3e5511a13c7ccae12038)

surajshetty3416 committed:
refactor:  Use null instead of undefined

(cherry picked from commit 1e3d58c14f614ecac1eb3e5511a13c7ccae12038)

surajshetty3416 committed:
Merge pull request #15327 from frappe/mergify/bp/version-13-pre-release/pr-15320

surajshetty3416 committed:
Merge pull request #15326 from frappe/mergify/bp/version-13-hotfix/pr-15320

surajshetty3416 committed:
Merge pull request #15322 from frappe/mergify/bp/version-13-hotfix/pr-14438

sagarvora committed:
fix: remove irrelevant comment

(cherry picked from commit 8b8f03f2d1f5b3c5cac38c93d45db2184ebe8634)

sagarvora committed:
perf: improve translation speed

(cherry picked from commit 774b1dcd07a4add4c4b96ac988039540bbbb8077)

# Conflicts:
#	frappe/public/js/desk.bundle.js

sagarvora committed:
fix: increase timeout to 100ms

(cherry picked from commit fdf060178b8c03ab70e7de52098b3c29d2738962)

sagarvora committed:
fix: dont create copy of object

(cherry picked from commit 0f98f8a3a4fa98cb644708b4c23c80f929008411)

saxenabhishek committed:
fix: misc setup_wizard region slide

placeholder on all fields
autocomplete on country input
default selected values on inputs

(cherry picked from commit 91bd5ed2e2a8413574ac26e0620157d59022cc73)

surajshetty3416 committed:
Merge branch 'version-13-hotfix' into fix-searchbar-translations

surajshetty3416 committed:
style: Minor formatting fixes

sagarvora committed:
fix: update build.json

sagarvora committed:
Merge pull request #15345 from frappe/mergify/bp/version-13-hotfix/pr-15337

perf: improve translation speed (backport #15337)

mergify[bot] committed:
Merge pull request #15305 from vrslev/fix-searchbar-translations

fix(Translations): Search suggestions for lists, reports etc

mergify[bot] committed:
fix: more button in multi-select dialog (backport #15312) (#15356)

Co-authored-by: Saqib Ansari <nextchamp.saqib@gmail.com>

mergify[bot] committed:
fix: value associated with property may be changed (backport #15352) (#15358)

Co-authored-by: hrwx <himanshuwarekar@yahoo.com>

mergify[bot] committed:
feat: sanitise frappe.qb (backport #15180) (#15357)

Co-authored-by: saxenabhishek <as7122000@gmail.com>
Co-authored-by: Suraj Shetty <13928957+surajshetty3416@users.noreply.github.com>

gavindsouza committed:
fix(ui): Rendering forms created by disabled users

frappe.utils.capitalize doesn't exist anymore (has been very long). And it didn't do that
good of a job for setting fullnames anyway.

(cherry picked from commit f8356141a91c9e11e48da230f369bf80ae1ab1af)

gavindsouza committed:
fix(ui): Rendering forms created by disabled users

frappe.utils.capitalize doesn't exist anymore (has been very long). And it didn't do that
good of a job for setting fullnames anyway.

sagarvora committed:
fix: typo in comment

Co-authored-by: Faris Ansari <netchampfaris@users.noreply.github.com>
(cherry picked from commit 6dc67835c5eb3497498432181590c55865934aaa)

mergify[bot] committed:
fix: treeview expanding at random (backport #15287) (#15372)

Co-authored-by: Ankush Menat <ankush@frappe.io>

mergify[bot] committed:
test: Fix flaky test of language selection (backport #15373) (#15375)

Co-authored-by: Suraj Shetty <13928957+surajshetty3416@users.noreply.github.com>

ankush committed:
feat: SQL savepoints

(cherry picked from commit 1c0de776340177fd76d1c47cf019eea01336539a)

ankush committed:
fix: return self when using `save` on new docs

for consistency.

(cherry picked from commit 3101c45f2680e3ddc71828c058a12c2b1721cc8c)

surajshetty3416 committed:
refactor: Use placeholder instead of additional option in select

- Additional options were still selectable which could have got through
- Minor formatting changes

(cherry picked from commit 1dd870ce73457b1505e74c706be14ea8a7a33867)

surajshetty3416 committed:
revert: Empty value checks

(cherry picked from commit b530beb503c0f70422daafb98817741c24ffd21e)

gavindsouza committed:
fix: Barcode rendering in server side printing (#15383)

Co-authored-by: Suraj Shetty <surajshetty3416@gmail.com>

mergify[bot] committed:
fix: Barcode rendering in server side printing (backport #15383) (#15385)

Co-authored-by: Suraj Shetty <surajshetty3416@gmail.com>
Co-authored-by: gavin <gavin18d@gmail.com>

mergify[bot] committed:
fix: default to 'en' if no language in settings (backport #15381) (#15386)

Co-authored-by: Conor <cpdeethree@users.noreply.github.com>

mergify[bot] committed:
feat: `before_uninstall` and `after_uninstall` hooks (backport #15339) (#15387)

Co-authored-by: phot0n <ritwikpuri5678@gmail.com>

sagarvora committed:
Merge pull request #15388 from frappe/mergify/bp/version-13-hotfix/pr-15318

feat: translations in Data Import (backport #15318)

mergify[bot] committed:
Merge pull request #15371 from gavindsouza/disabled-users-boot

fix(ui): Rendering forms created by disabled users

ankush committed:
Merge pull request #15389 from frappe/mergify/bp/version-13-hotfix/pr-15379

feat: SQL savepoints (backport #15379)

rohitwaghchaure committed:
chore: change log for v13.17.0

rohitwaghchaure committed:
Merge pull request #15397 from rohitwaghchaure/change-log-for-v13-17

chore: change log for v13.17.0

rohitwaghchaure committed:
Merge branch 'version-13-pre-release' into version-13

rohitwaghchaure committed:
bumped to version 13.17.0

adityahase committed:
fix(restore): Repopulate db_tables cache after database restore

This takes care of the following scenario.

During restore, before tabDefaultValue is created if a request/command executes frappe.db.get_tables, then the cached value db_tables is populated with partial list of tables.

bootstrap_database then checks if this partial list contains tabDefaultValue (as a sanity test to see if the restore worked). This check fails even though the database is correctly restored.

(cherry picked from commit d334379b02a1b776895cdf5621c5f9ae9604e74a)

mergify[bot] committed:
Merge pull request #15406 from frappe/mergify/bp/version-13-hotfix/pr-15404

fix(restore): Repopulate db_tables cache after database restore (backport #15404)

mymi14s committed:
Fixed Report boilerplate (js, py) year to current year

(cherry picked from commit cc2c071cb98082291a3069d9d3f8d10007c98b51)

rohitwaghchaure committed:
Merge pull request #15424 from frappe/mergify/bp/version-13-pre-release/pr-15371

fix(ui): Rendering forms created by disabled users (backport #15371)

rohitwaghchaure committed:
Merge branch 'version-13-pre-release' into version-13

rohitwaghchaure committed:
bumped to version 13.17.1

ankush committed:
Merge pull request #15428 from frappe/mergify/copy/version-13-hotfix/pr-15422

fix: set year to current year in Script Report boilerplate (copy #15422)

mergify[bot] committed:
feat: add consecutive calendar week (WW) for naming series (backport #14514) (#15412)

Co-authored-by: Daniel Gerhardt <code@dgerhardt.net>

mergify[bot] committed:
fix: mutating route history while get_last_doc (backport #15427) (#15430)

Co-authored-by: @skjbulcher
Co-authored-by: Ankush Menat <ankush@frappe.io>

hrwX committed:
fix: use python _ instead of __ (#15414)

Co-authored-by: Raffael Meyer <14891507+barredterra@users.noreply.github.com>

ks093 committed:
fix: Incorrect timespan daterange calculation (#15369)

pateljannat committed:
fix: discussions component datetime format

mergify[bot] committed:
fix: don't run webhook in migrate (backport #15449) (#15455)

Co-authored-by: Raffael Meyer <14891507+barredterra@users.noreply.github.com>

mergify[bot] committed:
fix: Grammar of error statement in 404 page (backport #15447) (#15456)

Co-authored-by: Suraj Shetty <13928957+surajshetty3416@users.noreply.github.com>
Co-authored-by: Wolfram Schmidt <wolfram.schmidt@tueit.de>

mergify[bot] committed:
fix(print page): additional menu in mobile view print page (backport #15435) (#15459)

Co-authored-by: karthikeyan5 <skarthikeyan1410@gmail.com>

mergify[bot] committed:
fix: Rename Property Setter on Print Format Rename (backport #15458) (#15461)

Co-authored-by: Gavin D'souza <gavin18d@gmail.com>

mergify[bot] committed:
Merge pull request #15453 from pateljannat/date-format

fix: discussions component datetime format

mergify[bot] committed:
feat: option to not rebuild search on rename (backport #15443) (#15462)

Co-authored-by: barredterra <14891507+barredterra@users.noreply.github.com>

mergify[bot] committed:
fix: Use `frappe.permissions.has_permission` for proper error message (backport #15467) (#15469)

Co-authored-by: Suraj Shetty <13928957+surajshetty3416@users.noreply.github.com>

hrwX committed:
fix: translations in setup wizard (#15401)

Co-authored-by: Raffael Meyer <14891507+barredterra@users.noreply.github.com>
(cherry picked from commit 8947a542d1d3152dbf68c6fcc6b22286d01dd52b)

deepeshgarg007 committed:
fix: Route options not getting set while navigating from one report to another

rohitwaghchaure committed:
chore: Merge branch 'version-13-hotfix' into 'version-13-pre-release'

rohitwaghchaure committed:
Merge pull request #15478 from rohitwaghchaure/merge-hotfix-to-pre-release-for-13-18

chore: Merge branch 'version-13-hotfix' into 'version-13-pre-release'

pruthvi145 committed:
fix: clear cache when filters passed as dn param in `set_value`

(cherry picked from commit 289d7e7afa8c922d9069579d3a898ba98b529a9b)

surajshetty3416 committed:
fix: Check if value for `applicable_for` exists before setting it null

To avoid "Not Saved" state even after saving the document

(cherry picked from commit b77d08d26a3ab6e3a1192f9fc6315c23c860174e)

surajshetty3416 committed:
fix: Check if value for `applicable_for` exists before setting it null

To avoid "Not Saved" state even after saving the document

(cherry picked from commit b77d08d26a3ab6e3a1192f9fc6315c23c860174e)
(cherry picked from commit 61c2e4a79256daa741c4bc6846f07b86fbe634ad)

surajshetty3416 committed:
refactor: Re-arraange fields in user permission to avoid confusion

(cherry picked from commit b2d16e7db8af27319bcbeed37e594483dabafe9c)

surajshetty3416 committed:
refactor: Re-arraange fields in user permission to avoid confusion

(cherry picked from commit b2d16e7db8af27319bcbeed37e594483dabafe9c)
(cherry picked from commit a58fe1a9e7434636ff089b26e6c57f094fa2d8f5)

surajshetty3416 committed:
refactor: Re-arrange user form

- To fix order of fields in quick entry form

(cherry picked from commit 0868ad02f84ce5a3593fa9d89fbcf597887ac6fc)

# Conflicts:
#	frappe/core/doctype/user/user.json

deepeshgarg007 committed:
fix: Linting Issues

surajshetty3416 committed:
Merge pull request #15499 from frappe/mergify/bp/version-13-hotfix/pr-15497

surajshetty3416 committed:
Merge pull request #15500 from frappe/mergify/bp/version-13-pre-release/pr-15499

surajshetty3416 committed:
fix: Start week from Sunday instead of Monday

- Like we do in client-side

(cherry picked from commit b5e2f78a883d13e5acd477130aab7dc59e1fc0c2)

surajshetty3416 committed:
test: Add cases to check date utilities

(cherry picked from commit 85360ef4c0c4204cf79c6248e122924993310f19)

# Conflicts:
#	frappe/tests/test_utils.py

surajshetty3416 committed:
fix: Do not allow rebuilding of tree for doctypes without `lft` & `rgt`

(cherry picked from commit 9258941151616c7099c09564975a70c9cd3bc642)

surajshetty3416 committed:
fix: Hide "Rebuild Tree" option if treeview is loaded for non-nestedset doctype

(cherry picked from commit d1ca3fa78c69aefb3da20470cc0537fc83e0b179)

surajshetty3416 committed:
fix: Use copy of an array while adding new content for proper change trigger

(cherry picked from commit 3e009e9bdb95ab8b4f684e4bc8a30829fb8ae6b5)

nextchamp-saqib committed:
fix: renaming a document updates the modified timestamp

(cherry picked from commit b0b19d7a09d6233e0b14c28b805fe893c528ff99)

ankush committed:
Merge pull request #15510 from frappe/mergify/bp/version-13-hotfix/pr-15508

fix: renaming a document updates the modified timestamp (backport #15508)

ankush committed:
Merge pull request #15512 from frappe/mergify/bp/version-13-hotfix/pr-15504

fix: Do not allow rebuilding of tree for doctypes without `lft` & `rgt` (backport #15504)

surajshetty3416 committed:
Merge pull request #15513 from frappe/mergify/bp/version-13-hotfix/pr-15507

surajshetty3416 committed:
fix: Resolve conflicts

surajshetty3416 committed:
Merge pull request #15514 from frappe/mergify/bp/version-13-hotfix/pr-15498

mergify[bot] committed:
Merge pull request #15477 from deepeshgarg007/query_report_route_options

fix: Route options not getting set while navigating from one report to another

surajshetty3416 committed:
feat: Add setting to configure the day on which week starts

(cherry picked from commit 42e1c15c18f72e447f1b7b5e4227963abbc90249)

# Conflicts:
#	frappe/core/doctype/system_settings/system_settings.json
#	frappe/public/js/frappe/desk.js

surajshetty3416 committed:
fix: Set start of the week for date control based on configuration

(cherry picked from commit 9c75336ea0e11d31adcf23b129f351d148b2569a)

surajshetty3416 committed:
test: Update date utils test cases

(cherry picked from commit 4a5d9a02e3b74b5c430f45514dc6fc5ccb3b7aa8)

surajshetty3416 committed:
fix: Add patch to set week_starts_on as "Monday"

(cherry picked from commit ca9b2a953d13a3c2b67a37c9aaba2da0d93fb8cc)

# Conflicts:
#	frappe/patches.txt

surajshetty3416 committed:
test: Fix dashboard chart tests

(cherry picked from commit 5821a7bc9176ad8fb11d39bf89212d29066f327d)

surajshetty3416 committed:
style: Fix formatting issues

(cherry picked from commit ea2101d537cab6541f047f9fe1879f8303474a5a)

surajshetty3416 committed:
fix: Revert unnecessary code

(cherry picked from commit 6b77ed68e9b74b85cf5bbee1c05630a8e36d2ebb)

surajshetty3416 committed:
fix: Re-setup moment object whenever `week_starts_on` is changed

(cherry picked from commit 53b2c472a6f4576ef7fbeea478d54b1ec54ba7ae)

# Conflicts:
#	frappe/core/doctype/system_settings/system_settings.js

mergify[bot] committed:
feat(minor): copy link to clipboard (backport #15523) (#15524)

Co-authored-by: Ankush Menat <ankush@frappe.io>

mergify[bot] committed:
Merge pull request #15530 from frappe/mergify/bp/version-13-hotfix/pr-15492

fix: clear cache for all docs matched with the filters passed as `dn` param in `set_value` (backport #15492)

surajshetty3416 committed:
refactor: Rename `week_starts_on` to `first_day_of_the_week`

For consistency

(cherry picked from commit 5d5ad78789738e6c9ddd3a368f44badc9022ce5a)

mergify[bot] committed:
fix(ldap): Auto-negotiate the highest TLS protocol version supported by client & server (backport #15517) (#15535)

Co-authored-by: Christoph Kappel <christoph.kappel@bytesource.net>

mergify[bot] committed:
fix: set `first_response_time` only if communication is sent (backport #15533) (#15536)

Co-authored-by: Saqib Ansari <nextchamp.saqib@gmail.com>

nextchamp-saqib committed:
feat: `frappe.enqueue` and `frappe.call` for server scripts (#15528)

Co-authored-by: Sagar Vora <sagar@resilient.tech>

gavindsouza committed:
fix: Use params instead of data for client login

(cherry picked from commit 984420363a9aa8d60f95b197e0750e1766347ddf)

# Conflicts:
#	frappe/frappeclient.py

gavindsouza committed:
Merge pull request #15540 from frappe/mergify/bp/version-13-hotfix/pr-14323

chore: Bump psycopg2-binary Python client (backport #14323)

barredterra committed:
fix: awesomeplete line wrapping

(cherry picked from commit bf28199a6dd14c63ac7feb0ec1ace67eeff57408)

surajshetty3416 committed:
test: UI test cases for "First Day of the Week"

(cherry picked from commit 54fbd0b5e6239398e0833557e718ff5bf96d578d)

surajshetty3416 committed:
style: Fix sider issues

(cherry picked from commit ed2e37eaf7c61e0687d9d817505b10b2654c8f4a)

surajshetty3416 committed:
test: Fix flaky grid flaky test

Simplified test case as bit

(cherry picked from commit 30591720f868e924973c2fd458448752d8a5295c)

surajshetty3416 committed:
test: Remove unnecessary code

(cherry picked from commit 454a08aa183d81cbaed1c3ff789a5943c92ddd82)

mergify[bot] committed:
fix(Migrate): add unique and index for new columns (backport #15484) (#15545)

Co-authored-by: Saif Ur Rehman <saifi0102@gmail.com>

gavindsouza committed:
fix: Use format instead of f-strings, resolve conflicts

gavindsouza committed:
Merge pull request #15546 from frappe/mergify/bp/version-13-hotfix/pr-15542

fix: FrappeClient post_api request being redirected to GET request (backport #15542)

nextchamp-saqib committed:
fix: catch exceptions while parsing json on request failure

(cherry picked from commit 55178101e85ec80b367cd88f84e251c4278c61ca)

surajshetty3416 committed:
test: Fix get_field command for select fields

(cherry picked from commit 0a8b5fe7857c875c6b08b728887d93ba96dd2cd4)

ahmadpak committed:
fix: email_string parsing when pulling email

(cherry picked from commit 6afa5c8da3c7d0e74425f572e39f042e76ae6d19)

# Conflicts:
#	frappe/core/doctype/communication/communication.py

surajshetty3416 committed:
fix: Resolve conflicts

ahmadpak committed:
refactor: communication.py

(cherry picked from commit 9ebfc1ced9cfb7643560858553e59243eb128ac4)

surajshetty3416 committed:
Merge pull request #15555 from frappe/mergify/bp/version-13-hotfix/pr-15502

rohitwaghchaure committed:
chore: change log for v13.18.0

rohitwaghchaure committed:
Merge pull request #15557 from rohitwaghchaure/change-log-for-v13-18

chore: change log for v13.18.0

rohitwaghchaure committed:
bumped to version 13.18.0

rohitwaghchaure committed:
Merge branch 'version-13-pre-release' into version-13

surajshetty3416 committed:
Merge branch 'version-13-hotfix' into mergify/bp/version-13-hotfix/pr-15401

surajshetty3416 committed:
fix: translations in setup wizard (backport #15401) (#15473)

Co-authored-by: Raffael Meyer <14891507+barredterra@users.noreply.github.com>
Co-authored-by: Himanshu <himanshuwarekar@yahoo.com>
Co-authored-by: Suraj Shetty <13928957+surajshetty3416@users.noreply.github.com>

surajshetty3416 committed:
Merge pull request #15558 from frappe/mergify/bp/version-13-hotfix/pr-15543

nextchamp-saqib committed:
chore: check custom exception handlers only for json response

(cherry picked from commit 307c7ced69572ec56051f3a88e1e91422d2e94fb)

nextchamp-saqib committed:
fix: parsing exception from traceback

(cherry picked from commit 1bde3b48bdd7a0d81f6d183b19f58ac8ecb8b72c)

ankush committed:
revert: "fix: translations in setup wizard (backport #15401)" (#15572)

pruthvi145 committed:
fix: limit in `Document.get`

(cherry picked from commit d1852242a0acb37c799d00c2b9fc8ca6d71b3587)

# Conflicts:
#	frappe/model/base_document.py

pruthvi145 committed:
test: add test for limit

(cherry picked from commit a87899c2155a0c2576d551a31aa5f6535be6930b)

sumaiya2908 committed:
fix: email fetch error for imaplib (#15581)

* fix: remove maxline for imaplib

* refactor: remove redundant line

Co-authored-by: Summayya <frappe@Summayyas-MacBook-Air.local>
(cherry picked from commit 645fde2e675daf0fd7e13ad44a051a5453c2577c)

# Conflicts:
#	frappe/email/receive.py

mergify[bot] committed:
Merge pull request #15577 from frappe/mergify/bp/version-13-hotfix/pr-15564

fix: communication method "get_contacts" (backport #15564)

mergify[bot] committed:
Merge pull request #15583 from frappe/mergify/bp/version-13-hotfix/pr-15581

fix: email fetch error for imaplib (backport #15581)

mergify[bot] committed:
Merge pull request #15585 from frappe/mergify/bp/version-13-hotfix/pr-15574

fix: limit in `Document.get` (backport #15574)

mergify[bot] committed:
fix: enable languages inserted during setup (#15588) (#15590)

(cherry picked from commit eeff2a22f88b8f836a4cf0bc5935730318f58108)

Co-authored-by: Ankush Menat <ankush@frappe.io>

mergify[bot] committed:
Merge pull request #15595 from frappe/mergify/bp/version-13-hotfix/pr-15549

fix: catch exceptions while parsing json on request failure (backport #15549)

surajshetty3416 committed:
fix: Add toggle theme as standard dropdown option

- For new sites toggle theme was not available. It was added via patch for old sites.

(cherry picked from commit 645f23a9a7c28dfa0271888e419ff13b4b4b2986)

surajshetty3416 committed:
Merge pull request #15611 from frappe/mergify/bp/version-13-hotfix/pr-15610

surajshetty3416 committed:
Merge pull request #15573 from frappe/mergify/bp/version-13-hotfix/pr-15277

mergify[bot] committed:
fix: AttributeError when calling frappe.permissions.$function (backport #15615) (#15616)

Co-authored-by: Ankush Menat <ankush@frappe.io>

mergify[bot] committed:
fix: patch for validating options field (#15565) (#15623)

* fix: patch for validating options field

* fix: reorder patch execution

* refactor: reduce queries for patching

* refactor: change patch name and license

Co-authored-by: Ankush Menat <ankush@frappe.io>
(cherry picked from commit c3409c818eba2e0d4e30bf27a8bd8aabadd4c566)

Co-authored-by: Shadrak Gurupnor <30501401+shadrak98@users.noreply.github.com>

mergify[bot] committed:
fix: translations in setup wizard (#15401) (#15633)

Co-authored-by: Raffael Meyer <14891507+barredterra@users.noreply.github.com>
(cherry picked from commit 8947a542d1d3152dbf68c6fcc6b22286d01dd52b)

Co-authored-by: Himanshu <himanshuwarekar@yahoo.com>

rohitwaghchaure committed:
chore: Merge branch 'version-13-hotfix' into 'version-13-pre-release'

rohitwaghchaure committed:
Merge pull request #15642 from rohitwaghchaure/merge-hotfix-to-pre-release-for-13-19

chore: Merge branch 'version-13-hotfix' into 'version-13-pre-release'

mergify[bot] committed:
fix: validate_link calls via Edit Shortcut in Dashboard (backport #15657) (#15658)

Co-authored-by: Summayya <frappe@Summayyas-MacBook-Air.local>
Co-authored-by: Shariq Ansari <sharique.rik@gmail.com>

mergify[bot] committed:
fix: Set first day of the week for date range (backport #15669) (#15672)

Co-authored-by: Suraj Shetty <13928957+surajshetty3416@users.noreply.github.com>

rohitwaghchaure committed:
chore: change log for v13.19.0

rohitwaghchaure committed:
Merge pull request #15689 from rohitwaghchaure/change-log-for-v13-19

chore: change log for v13.19.0

rohitwaghchaure committed:
bumped to version 13.19.0

rohitwaghchaure committed:
Merge branch 'version-13-pre-release' into version-13

e-lobo committed:
fix: import file

e-lobo committed:
Merge branch 'version-13' into version-13-fixes-v13-15-0

# Conflicts:
#	frappe/modules/import_file.py

e-lobo commented:
All good here @assemmarwan @fahimalizain @Abadulrehman

Abadulrehman commented:
Looks good. 👍
